Scoring Maintainence

howard-beck opened the pull request:
Added more comments, redistributed some method groupings, removed unused methods, privatized some methods that should only be used within the class, made encoder ticks = 0 on arm mean its 100% up, any positive encoder reading is front of robot etc (hard to understand what was going on before when switching between orientations), made switching the orientation of the arm... actually switch the orientation of the arm instead of waiting until another level state to reposition the arm

Commit:
Fixed scoring state machine (again)

it b like that sometimes

If you pressed 2 level buttons and switch the orientation, it would previously switch the orientation. Now, we have 2 more levels, for no selected one and invalid (multiple selected) to fix this; null should not mean 2 different things

Commit:
Removed unused getAnySelectedLevel()

Commit:
Merge remote-tracking branch 'BitBucketsFRC4183/master'

Commit:
Scoring Maintenance 

Added more comments, redistributed some method groupings, removed unused methods, privatized some methods that should only be used within the class, made encoder ticks = 0 on arm mean its 100% up, any positive encoder reading is front of robot etc (hard to understand what was going on before when switching between orientations), made switching the orientation of the arm... actually switch the orientation of the arm instead of waiting until another level state to reposition the arm

Commit:
Scoring Maintainence

Added more comments, redistributed some method groupings, removed unused methods, privatized some methods that should only be used within the class, made encoder ticks = 0 on arm mean its 100% up, any positive encoder reading is front of robot etc (hard to understand what was going on before when switching between orientations), made switching the orientation of the arm... actually switch the orientation of the arm instead of waiting until another level state to reposition the arm

Commit:
Added timeouts on scoring commands

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
BB_Motor utility class work

Continued working on BB_Motor utility class. Some changes we will need to make: fix a idx for position and velocity control to put methods like getPosition_inch(), getAngle_inch(), etc...

Commit:
Added force Idle button and fixed timeouts

cpostbitbuckets commented on DestinationDeepSpace/src/main/java/frc/robot/subsystem/scoring/ScoringSubsystem.java:
```diff
@@ -288,7 +311,24 @@ public void initialize() {
 		SmartDashboard.putNumber(getName() + "/Test Angle", 0);
 	}
 
+
```
Look at all those spaces!

cpostbitbuckets commented on DestinationDeepSpace/src/main/java/frc/robot/utils/talonutils/BB_Motor.java:
```diff
@@ -16,29 +19,42 @@
     // with a scoring arm, for example, it would be the length of the scoring arm
-    private double radius;
-    private double circumference;
+    private double radius_inch;
```
standard java naming would be more like `radiusInches`

RocketRedNeck replied:
Do we want to establish that as a standard? I use the _unit nomenclature in C and C++ to ensure that the unit really stands out.

Commit:
XBox trackpad, fixed timeouts/force idle, worked on AutoTuner

AutoTuner:
- changed data to doubles (to include power %)
- will add in DFT calculation soon
- removed getOscillations (more sophisticated oscillation algorithm coming soon)
- every TuningStep now officially collections Position or Velocity data
- added automatic stability detection for velocity and position (using data collection types added to decide what threshold to use)
- removed unused import
- all tuning steps now have power datas too

Commit:
Added automatic DFT to AutoTuner DataWindow

It will automatically change the complex amplitudes when data is added using an efficient algorithm. When we get a feel for what an "oscillating" frequency spectrum is, I'll replace this with a FFT that gets run every few iterations (as long as its more efficient than the current algorithm every iteration)

Commit:
Merge remote-tracking branch 'BitBucketsFRC4183/master'

Commit:
Added AutoTuner step comments
